src/templates/contest/contest.html

```diff
@@ -45,6 +46,11 @@ <h1 style="display: inline;">{{ title }}</h1>
                 <td><a href="{{ request.path }}/problem/{{ row["problem_id"] }}">{{ row["name"] }}</a></td>
                 <td>{{ row["category"] }}</td>
                 <td>{{ row["point_value"] }}</td>
+                {% if session.admin %}
+                    <td><a href="/admin/submissions?problem_id={{ row["id"] }}&contest_id={{ row["contest_id"] }}&correct=AC">{{ row["sols"] }}</a></td>
```

Comment:
Suggestion: replace `row["contest_id"]` with `request.path[9:]` and remove `"contest_id": row["problem_id"],` (application.py:499) to optimize code since the contest_id never changes between problems